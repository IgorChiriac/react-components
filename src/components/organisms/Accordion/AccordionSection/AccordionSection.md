### Summary

<Accordion.Section /> is a simple block element that passes down its `ref` to its children.
In this way, we can allow the section to be collapsible.

- It's meant to be used along `<Accordion.Section>` and `useAccordion`.
- [`<Accordion />`](/#/Components/Organisms/Accordion) examples for how they're all meant to play together.
