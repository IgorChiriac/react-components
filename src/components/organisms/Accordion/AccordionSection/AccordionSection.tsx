import React from 'react';

export interface IAccordionSection extends React.HTMLAttributes<HTMLDivElement> {}

const AccordionSection: React.FunctionComponent<IAccordionSection> = React.forwardRef<
  HTMLDivElement,
  IAccordionSection
>(({ children, ...rest }, ref) => (
  <div {...rest}>
    <div ref={ref}>{children}</div>
  </div>
));

export default AccordionSection;
