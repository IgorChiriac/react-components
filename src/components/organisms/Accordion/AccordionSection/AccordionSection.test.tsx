import { axe } from 'jest-axe';
import React from 'react';
import { render } from 'react-testing-library';
import AccordionSection from './AccordionSection';

describe('<AccordionSection />', () => {
  it('renders the component with no a11y violations', async () => {
    const { container } = render(<AccordionSection>Content</AccordionSection>);
    expect(container.firstChild).toMatchSnapshot();
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
});
