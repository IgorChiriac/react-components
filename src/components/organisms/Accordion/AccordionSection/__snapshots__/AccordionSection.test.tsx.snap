// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionSection /> renders the component with no a11y violations 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="one"
  data-testid="section"
  id="one-tab"
  role="region"
  style="overflow: hidden; transition: height 200ms linear; height: 0px;"
>
  <div>
    Content
  </div>
</div>
`;
